### PR TITLE
Include simple and intuitive RunnableDemo example

### PR DESCRIPTION
**Pull request by pmcb55**
# New feature description

Included a new RunnableDemo example, which very quickly and easily demonstrates some of the big features of the AG with a simple, intuitive Node.js script that uses simple Questions and Answers to show vocab term and metadata usage.
(Had to make tiny modification to `debug` code to remove noisy timing output, but we nicely color-coded output as a result (if you make your terminal window wide enough to avoid line-wrapping!)

# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] The changelog has been updated, if applicable.
- [X] New functions/types have been exported in `index.ts`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: example/RunnableDemo/Vocab/sample-vocab-bundle.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+# The artifact name is shared among all the artifacts.
+artifactName: ag-demo
+artifactGeneratorVersion: 2.0.0
+
+artifactToGenerate:
+  - programmingLanguage: Java
+    artifactVersion: 0.1.0
+    artifactNamePrefix: ""
+    artifactNameSuffix: ""
```
Comment (NSeydoux) [inline example/RunnableDemo/Vocab/sample-vocab-bundle.yml:15]: If left empty, should these be included at all?

File: example/RunnableDemo/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "ag-demo",
+  "version": "1.0.0",
```
Comment (NSeydoux) [inline example/RunnableDemo/package.json:3]:
```suggestion
  "name": "ag-demo",
  "private": true,
  "version": "1.0.0",
```

File: example/RunnableDemo/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "description": "Demo project for the Artifact Generator",
+  "main": "index.js",
+  "author": "Andy Davidson",
```
Comment (NSeydoux) [inline example/RunnableDemo/package.json:6]: I guess this is a copy/paste typo, and it should be formatted according to https://docs.npmjs.com/cli/v6/configuring-npm/package-json#people-fields-author-contributors

File: example/RunnableDemo/package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "license": "ISC",
```
Comment (NSeydoux) [inline example/RunnableDemo/package.json:7]:
```suggestion
  "license": "MIT",
```

Comment (pmcb55) [reply example/RunnableDemo/package.json:7]: Done: 92c6888

File: example/RunnableDemo/package.json (hunk 2 above)
Comment (pmcb55) [reply example/RunnableDemo/package.json:3]: Done: 7e45c0d

**Pull request merged**